### ISSUE TITLE
memoize with no arguments in or out doesn't work as expected

### ISSUE THREAD

**Issue by erg** [factor/factor#2780]
I expect `sync-iso-codes` to only hit the network once if I run it twice in a row. I just want to update this directory once per run or something. I could also check the timestamp on the directory but memo seems like it should work.

```factor
USING: cli.git io.files.temp io.pathnames json kernel ;
IN: iso-codes

: iso-codes-directory ( -- directory ) temp-directory "iso-codes" append-path ;

MEMO: sync-iso-codes ( -- )
    "https://salsa.debian.org/iso-codes-team/iso-codes.git"
    iso-codes-directory sync-repository-as drop ;

: iso-code-json ( path -- json )
    [ sync-iso-codes iso-codes-directory ] dip append-path path>json ;

: iso-639-2-json ( -- json ) "data/iso_639-2.json" iso-code-json ;
: iso-639-3-json ( -- json ) "data/iso_639-3.json" iso-code-json ;
: iso-639-5-json ( -- json ) "data/iso_639-5.json" iso-code-json ;



sync-iso-codes
sync-iso-codes

! really i want to do the following three lines and have them hit the network only once (or zero times if memo'd already)
iso-639-2-json
iso-639-3-json
iso-639-5-json
```

**Comment by bpollack** [comment]
I mean, we could just devolve this use case into `CONSTANT: foobar $[ ... ]`, but I'm curious why it happens.

**Comment by bpollack** [comment]
We handle 0 args specially through this whole call sequence. This looks fun.

**Comment by mrjbq7** [comment]
Fixed in a937766d2f1c0866fbe8433f3a121f346c8a33ca.
